### ISSUE TITLE
Feature | integrate and improve automated validation tests script

### ISSUE THREAD

**Issue by exequielrafaela** [binbashar/leverage#30]
## What?

- ✅ Include and improve validation tests [Leverage reference architecture script](https://github.com/binbashar/le-tf-infra-aws/blob/master/%40bin/scripts/validate-terraform-layout.sh)

- **Possible Checks** 
  - terraform fmt
  - terraform tflint
  - terraform validate
  - security (https://github.com/binbashar/le-tf-infra-aws/issues/73)
  - networking (overlapping, other validations?)

## Why?

- ✅ Validate that the terraform code canonical format, layout and security enforcing best practices. 
- ✅ Early phase error detection.

## Reference 

Current version is functional and could be found at the @bin folder in the Terraform Ref Arch repo.

```
╭─    ~/Binbash/repos/Leverage/ref-architecture/le-tf-infra-aws  on   master ····· ✔  at 12:32:08 
╰─ cd shared/base-network

╭─    ~/B/r/L/ref-architecture/le-tf-infra-aws/shared/base-network  on   master ·· ✔  at 12:32:13 
╰─ leverage validate_layout
[DEBUG] Found config file: /Users/exequielbarrirero/Binbash/repos/Leverage/ref-architecture/le-tf-infra-aws/shared/base-network/build.env
[DEBUG] Found config file: /Users/exequielbarrirero/Binbash/repos/Leverage/ref-architecture/le-tf-infra-aws/build.env
[ build.py - Starting task "_checkdir" ]
[ build.py - Completed task "_checkdir" ]
[ build.py - Starting task "validate_layout" ]
[INFO] CWD=/Users/exequielbarrirero/Binbash/repos/Leverage/ref-architecture/le-tf-infra-aws/shared/base-network
[INFO] REPO_DIR=/Users/exequielbarrirero/Binbash/repos/Leverage/ref-architecture/le-tf-infra-aws
[INFO] ACCOUNT_NAME: shared
[INFO] CATEGORY_NAME: base-network

[INFO] Checking if backend key starts with shared ...
[INFO]  -> OK
[INFO] Checking if backend key starts with shared and follows with base-network ...
[ERROR]  -> FAILED
[INFO] Checking if backend.config profile contains shared ...
[INFO]  -> OK
[INFO] Checking if backend.config bucket contains shared ...
[INFO]  -> OK
[INFO] Checking if backend.config dynamodb_table contains shared ...
[INFO]  -> OK
[INFO] Checking if base.config  contains shared ...
[INFO]  -> OK
[ build.py - Completed task "validate_layout" ]
```

**Comment by exequielrafaela** [comment]
@angelofenoglio @diego-ojeda-binbash as we've talked, I'm adding this new comment here to get this issue prioritized

### TLDR

- there should be a `tf` task to validate that the S3 state backend key is the correct one for the layer you are in.
- this task should then be pre-task of the `tf init`, maybe of other tasks as well (to be defined).

---

#### CONSIDERATIONS

- So If you are going to `leverage tf apply` locally we expect to have the pre-check validation automatically executed by default and interrupt the exec returning an error message.
- We need the check to be run before a `leverage tf [cmd]` through the CLI, as it is a validation that the cli could easily do.
- It could be run as a standalone command and at the same time before a `tf init`.  
- The more we shift left these type of validations the sooner we detect them, for instance before it is applied. 
- Consider `ex-pynt` allows us to do a task that is a pre-task of another one. Since we rely on this framework, the check command should be easy to develop and implement.

---

CC: @binbashar/leverage-ref-architecture-aws-admin @binbashar/leverage-ref-architecture-aws-dev
